### PR TITLE
Removed explicit capturing of "this" for lamdas

### LINKED ISSUE #166: Capturing "this" context
In the transport and connection typescript classes, you are capturing "this" explicitly but only using lamdas which will capture this for you. See [LongPollingTransport](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.SignalR.Client.TS/Transports.ts#L160). Is there any reason for that? I've [committed something](https://github.com/herecydev/SignalR/commit/98bc22aed75584fff7f41e2ea2d9c6640bdec6c1) if you would like a PR?

### PR DESCRIPTION
**Pull request by herecydev**
Resolves #166

### CODE REVIEW

File: src/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts [hunk 1]
```diff
@@ -68,18 +67,16 @@ export class HubConnection {
     }
 
     invoke(methodName: string, ...args: any[]): Promise<any> {
-
-        let id = this.id;
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts]: not sure about this change, won't the promise potentially capture the wrong value for id at https://github.com/aspnet/SignalR/pull/167/files#diff-98c1bfb809ef38f16f965a508fe52526R79

Comment (herecydev) [inline src/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts]: Yep, a little overzealous on the removal.

**Comment by moozzyk** [comment]
Is it correct though? My understanding is that in case of event handlers `this` will be set to `event.currentTarget` and this is the reason why in all event handlers I uses the `that = this` pattern.
image

**Comment by herecydev** [comment]
I'm pretty tired so forgive me if this is nonsense, but I've put together [this gist](https://gist.github.com/herecydev/294f1aedcf1bc3a58fc80652f8e68b79).

Either using this or explicitly capturing it results in the correct "this" being capture:

```js
var _this = this;
var thisHubConnection = this;
```

**Comment by moozzyk** [comment]
Just noticed my mistake - in the picture above I marked `this` but this was in the Local scope and not in the closure. I need to confirm that indeed `this` works as expected in arrow functions used as event handlers and if this is the case I will merge this PR. 
(btw. I ran end-to-end TS client tests with your change and all tests passed so most likely this change is good).

**Comment by moozzyk** [comment]
I think I was wrong and this should be captured just fine in event handlers.

**Pull request merged**